internal/bminventory/inventory.go

```diff
@@ -2215,17 +2216,18 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(cluster *common.Cluster,
 	if params.ClusterUpdateParams.IngressVips != nil {
 		targetConfiguration.IngressVips = params.ClusterUpdateParams.IngressVips
 	}
+	if params.ClusterUpdateParams.LoadBalancer != nil {
+		targetConfiguration.LoadBalancer = params.ClusterUpdateParams.LoadBalancer
+	}
 	if params.ClusterUpdateParams.MachineNetworks != nil &&
 		common.IsSliceNonEmpty(params.ClusterUpdateParams.MachineNetworks) &&
 		interactivity == Interactive &&
-		!reqDualStack {
-		err := errors.New("Setting Machine network CIDR is forbidden when cluster is not in vip-dhcp-allocation mode")
+		!reqDualStack &&
+		!network.IsLoadBalancerUserManaged(&targetConfiguration) {
+		err := errors.New("Setting the Machine Network CIDR is forbidden when the cluster is neither in VIP DHCP allocation mode nor using a user-managed load balance")
```

Comment:
Type, should be `balancer`, with the `r`.

Reply:
Thank you, I will patch this in a separate PR